R/verb-compute.R

```diff
@@ -81,6 +81,12 @@ collect.tbl_sql <- function(x, ..., n = Inf, warn_incomplete = TRUE, cte = FALSE
   }
 
   sql <- db_sql_render(x$src$con, x, cte = cte)
-  out <- db_collect(x$src$con, sql, n = n, warn_incomplete = warn_incomplete)
+  call <- current_env()
+  tryCatch(
+    out <- db_collect(x$src$con, sql, n = n, warn_incomplete = warn_incomplete),
+    error = function(cnd) {
+      cli_abort("Can't collect lazy table.", parent = cnd, call = call)
```

Comment:
Are you sure you need the `call` here?

Reply:
And something about the wording seems off to me? Maybe "failed to collect lazy table" or "can't execute SQL"?

Reply:
Nice, I don't need `call`. rlang really does a great job 😄